File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/render/HUD.java
package net.ccbluex.liquidbounce.features.module.modules.render;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.event.*;
import net.ccbluex.liquidbounce.features.module.Module;
import net.ccbluex.liquidbounce.features.module.ModuleCategory;
import net.ccbluex.liquidbounce.features.module.ModuleInfo;
import net.ccbluex.liquidbounce.ui.client.hud.designer.GuiHudDesigner;
import net.ccbluex.liquidbounce.value.BoolValue;
import net.minecraft.client.gui.GuiChat;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

/**
 * LiquidBounce Hacked Client
 * A minecraft forge injection client using Mixin
 *
 * @game Minecraft
 * @author CCBlueX
 */
@ModuleInfo(name = "HUD", description = "Toggles visibility of the HUD.", category = ModuleCategory.RENDER)
@SideOnly(Side.CLIENT)
public class HUD extends Module {

    public final BoolValue blackHotbarValue = new BoolValue("BlackHotbar", true);
    public final BoolValue inventoryParticle = new BoolValue("InventoryParticle", false);
    private final BoolValue blurValue = new BoolValue("Blur", false);
    public final BoolValue fontChatValue = new BoolValue("FontChat", false);

    public HUD() {
        setState(true);
        setArray(false);
    }

    @EventTarget
    public void onRender2D(Render2DEvent event) {
        if(LiquidBounce.CLIENT.hud == null || mc.currentScreen instanceof GuiHudDesigner)
            return;

        LiquidBounce.CLIENT.hud.render(false);
    }

    @EventTarget
    public void onUpdate(final UpdateEvent event) {
        LiquidBounce.CLIENT.hud.update();
    }

    @EventTarget
    public void onKey(final KeyEvent event) {
        LiquidBounce.CLIENT.hud.handleKey('a', event.getKey());
    }

    @EventTarget(ignoreCondition = true)
    public void onScreenChange(ScreenEvent event) {
        if(mc.theWorld == null || mc.thePlayer == null)
            return;

        if(getState() && event.getGuiScreen() != null && !(event.getGuiScreen() instanceof GuiChat || event.getGuiScreen() instanceof GuiHudDesigner) && blurValue.get())
            mc.entityRenderer.loadShader(new ResourceLocation(LiquidBounce.CLIENT_NAME.toLowerCase() + "/blur.json"));
        else
            mc.entityRenderer.stopUseShader();
    }
}

File: shared/main/java/net/ccbluex/liquidbounce/features/module/modules/render/HUD.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.render

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.event.*
import net.ccbluex.liquidbounce.features.module.Module
import net.ccbluex.liquidbounce.features.module.ModuleCategory
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import net.ccbluex.liquidbounce.ui.font.Fonts
import net.ccbluex.liquidbounce.value.BoolValue
import net.ccbluex.liquidbounce.value.FontValue
import net.ccbluex.liquidbounce.value.ValueGroup

/**
 * TODO
 * * Add more notification options
 */
@ModuleInfo(name = "HUD", description = "Toggles visibility of the HUD.", category = ModuleCategory.RENDER, array = false)
class HUD : Module()
{
    val blackHotbarValue = BoolValue("BlackHotbar", true)
    val inventoryParticle = BoolValue("InventoryParticle", false)
    private val blurValue = BoolValue("Blur", false)
    private val notificationGroup = ValueGroup("Notification")
    val notificationAlertsValue = BoolValue("Alerts", true, "Alerts")
    val notificationWorldChangeValue = BoolValue("WorldChange", true, "WorldChangeAlerts")
    val notificationModuleManagerValue = BoolValue("ModuleToggle", true) // TODO: ModuleToggleNotificationKeepTime
    val fontChatValue = BoolValue("FontChat", false)
    val chatFontValue = FontValue("FontChatFont", Fonts.font40)

    init
    {
        notificationGroup.addAll(notificationAlertsValue, notificationWorldChangeValue, notificationModuleManagerValue)
    }

    @EventTarget
    fun onRender2D(@Suppress("UNUSED_PARAMETER") event: Render2DEvent)
    {
        if (classProvider.isGuiHudDesigner(mc.currentScreen)) return

        LiquidBounce.hud.render(false)
    }

    @EventTarget
    fun onUpdate(@Suppress("UNUSED_PARAMETER") event: UpdateEvent)
    {
        LiquidBounce.hud.update()
    }

    @EventTarget
    fun onKey(event: KeyEvent)
    {
        LiquidBounce.hud.handleKey('a', event.key)
    }

    @EventTarget(ignoreCondition = true)
    fun onScreen(event: ScreenEvent)
    {
        mc.theWorld ?: return
        mc.thePlayer ?: return

        val entityRenderer = mc.entityRenderer

        val provider = classProvider

        val screen = event.guiScreen

        if (state && blurValue.get() && !entityRenderer.isShaderActive() && screen != null && !provider.isGuiChat(screen) && !provider.isGuiHudDesigner(screen)) entityRenderer.loadShader(provider.createResourceLocation(LiquidBounce.CLIENT_NAME.toLowerCase() + "/blur.json"))
        else
        {
            val shaderGroup = entityRenderer.shaderGroup

            if (shaderGroup != null && shaderGroup.shaderGroupName.contains("liquidbounce/blur.json")) entityRenderer.stopUseShader()
        }
    }

    init
    {
        state = true
    }
}
